src/Nethermind/Nethermind.Optimism/OptimismPlugin.cs

```diff
@@ -180,11 +180,14 @@ public async Task InitRpcModules()
         ArgumentNullException.ThrowIfNull(_invalidChainTracker);
         ArgumentNullException.ThrowIfNull(_blockFinalizationManager);
 
-        // Ugly temporary hack to not receive engine API messages before end of processing of all blocks after restart.
-        // Then we will wait 5s more to ensure everything is processed
-        while (!_api.BlockProcessingQueue.IsEmpty)
-            await Task.Delay(100);
-        await Task.Delay(5000);
+        if (!_api.Config<IInitConfig>().InRunnerTest)
+        {
+            // Ugly temporary hack to not receive engine API messages before end of processing of all blocks after restart.
+            // Then we will wait 5s more to ensure everything is processed
+            while (!_api.BlockProcessingQueue.IsEmpty)
+                await Task.Delay(100);
+            await Task.Delay(5000);
+        }
```

Comment:
Maybe this should be some extension method? It is in 3 places.

Reply:
I don't know why this is here at all.